src/hotspot/os/bsd/os_bsd.cpp

```diff
@@ -1634,7 +1634,7 @@ bool os::pd_uncommit_memory(char* addr, size_t size, bool exec) {
 }
 
 bool os::pd_create_stack_guard_pages(char* addr, size_t size) {
-  return os::commit_memory(addr, size, !ExecMem);
+  return os::pd_commit_memory(addr, size, !ExecMem);
 }
```

Comment:
Sorry but this is wrong IMO. A pd_ function should in general still call the os:: version of other functions because it doesn't know what that might do in addition to eventually calling the pd_ variant.